ILLink: Keep parameter names unless DebuggerSupport is disabled

sbomer opened the pull request:
Fixes https://github.com/dotnet/runtime/issues/102873

Parameter name trimming is currently the only metadata trimming optimization, but I put "all" under this check to include any future metadata trimming optimizations that might impact debuggers.

@dotnet/appmodel PTAL

sbomer committed:
Keep parameter names unless DebuggerSupport is disabled

MichalStrehovsky reviewed (approved):
Nit: did you consider adding a TrimmingTest? (Testing this could probably be approximated with trim-unsafe reflection and we could have variants for both DebuggerSupport=false and the default.)

sbomer committed:
Merge remote-tracking branch 'origin/main' into paramNameTrim

sbomer committed:
Add TrimmingTest

sbomer committed:
Set NativeAotIncompatible

sbomer committed:
Fix test

Removed parameter name is empty, not null, on wasm
Also remove unnecessary WriteLine
